vignettes/remind2_reporting.Rmd

```diff
@@ -0,0 +1,92 @@
+---
+title: "remind2 reporting"
+author: "Felix Schreyer"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{remind2 reporting}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, echo = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+
+## Function
+
+The ``remind2`` library contains the routines to generate the reporting of the output of the REMIND model and is the successor of the deprecated ``remind`` package since REMIND version 3.0. It's main function is to create the output of REMIND as a table in IAM-comparable format, the so-called ``mif-file``, from the ``fulldata.gdx`` of a run. The resulting ``mif-file`` is typically called ``REMIND_generic_NameOfYourRun.mif`` and can be found in the output folder of a run. 
+
+This REMIND reporting based on a GDX-file can be created with the function ``convGDX2MIF()``
+
+
+```{r}
+gdx <- "fulldata.gdx"
+testReport <- convGDX2MIF(gdx)
+```
+
+
+The calculation of the REMIND variables as reported in the ``mif-file`` is separated into different functions per variable category. For example, ``SE|...`` variables are calculated in reportSE.R, ``Emi|...`` variables in reportEmi.R etc. If you only want to check the calculation of a specific REMIND variables, you can call those functions directly:
+
+
+```{r}
+gdx <- "fulldata.gdx"
+testFE <- reportFE(gdx)
+```
+
+
+
+## Development
+
+
+
+When developing the remind2 library, please test your changes on a recent ``fulldata.gdx`` file from REMIND by running the ``convGDX2MIF()`` function. Make sure the ``fulldata.gdx`` file is suitable for testing the change you did. For example, check that the relevant GAMS variables and parameters are available and non-zero in the GDX you are testing. Different REMIND runs come with different module realization and switches where certain sets, parameters or variables are not existing such that your change may not be tested by an unsuitable GDX. Moreover, when adapating or adding variables with "+" notation, please run a summation check on a suitable GDX by running the ``checkIntegrity()`` found [here](https://github.com/pik-piam/remind2/blob/1a80cc607db663a211f98cabb3c7408cd08bf713/tests/testthat/test-convGDX2mif.R#L31) on your reporting output:
+
+
+```{r}
+checkIntegrity(testReport)
+```
+
+
+If your change involves reading in a parameter, variable or set from the GDX-file that is not yet merged to the [main REMIND repository](https://github.com/remindmodel/remind) or that is not declared in the default runs of REMIND, please use conditional statements that check whether these parameters exist. Otherwise, the reporting will break for other users with other GDX files. 
```

Comment:
Not sure what you mean.  Anything read in that is not in [rse.pik-potsdam.de/data/example/remind2_test-convGDX2MIF_fulldata.gdx](https://github.com/pik-piam/remind2/blob/1d3c4f98cb7e55d4769723b13da0a62db33b0894/tests/testthat/test-convGDX2mif.R#L58-L59) should throw an error and not even make it to the RSE package repository.

Reply:
If the ``fulldata.gdx`` downloaded is very old, then it might not trigger the if-clauses of your run that is some weeks old. But, in general Michaja is right. 

The trade-off of a lot of if-clauses which never get cleaned-up is that the code becomes harder to read.